src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler.php

```diff
@@ -403,27 +403,58 @@ private static function hasColorSupport()
             return false;
         }
 
-        if ('Hyper' === getenv('TERM_PROGRAM')) {
+        if (!self::isTty()) {
             return true;
         }
 
-        if (\DIRECTORY_SEPARATOR === '\\') {
-            return (\function_exists('sapi_windows_vt100_support')
-                && sapi_windows_vt100_support(\STDOUT))
-                || false !== getenv('ANSICON')
-                || 'ON' === getenv('ConEmuANSI')
-                || 'xterm' === getenv('TERM');
+        if ('\\' === \DIRECTORY_SEPARATOR
+            && \function_exists('sapi_windows_vt100_support')
+            && @sapi_windows_vt100_support(\STDOUT)
+        ) {
+            return true;
         }
 
+        if ('Hyper' === getenv('TERM_PROGRAM')
+            || false !== getenv('COLORTERM')
+            || false !== getenv('ANSICON')
+            || 'ON' === getenv('ConEmuANSI')
+        ) {
+            return true;
+        }
+
+        if ('dumb' === $term = (string) getenv('TERM')) {
+            return false;
+        }
+
+        // See https://github.com/chalk/supports-color/blob/d4f413efaf8da045c5ab440ed418ef02dbb28bf1/index.js#L157
+        return preg_match('/^((screen|xterm|vt100|vt220|putty|rxvt|ansi|cygwin|linux).*)|(.*-256(color)?(-bce)?)$/', $term);
```

Comment:
shouldn't it be `@\`? (not sure if the `pcre` extension is declared in the `composer.json` either

Reply:
why would it? preg_match doesn't trigger notices unless the regexp is badly formatted, which it isn't
about ext-pcre, we never list it as a requirement - its one of the builtin extensions.